src/binary-reader.cc

```diff
@@ -672,8 +591,17 @@ Result BinaryReader::ReadAddress(Address* out_value,
 }
 
 Result BinaryReader::ReadFunctionBody(Offset end_offset) {
+  return ReadInstructions(end_offset);
+}
+
+//
+// end_offset == UINT32_MAX means read until and end
+//  instruction is seen.
+// end_offset < UINT32_MAX means read until a given location
+//
+Result BinaryReader::ReadInstructions(Offset end_offset) {
```

Comment:
overloading `end_offset` seems a bit hacky. Is that better or worse than adding another parameter such as `single_expr` that just means 'return when you hit an `end` (and in that case if you get to the `end_offset` before hitting the end, it could be an error)